lightning/src/ln/channel.rs

```diff
@@ -3771,6 +3831,52 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 		self.channel_transaction_parameters.channel_type_features = self.channel_type.clone();
 		Ok(())
 	}
+
+	// /// Splice process starting; update state; update capacity, state, reset funding tx
+	// #[cfg(splicing)]
+	// pub(crate) fn splice_start<L: Deref>(&mut self, is_outgoing: bool, logger: &L) where L::Target: Logger {
+	// 	// Set state, by this point handshake is complete
+	// 	self.channel_state = ChannelState::NegotiatingFunding(NegotiatingFundingFlags::OUR_INIT_SENT | NegotiatingFundingFlags::THEIR_INIT_SENT);
+
+	// 	log_info!(logger,
+	// 		"Splicing process started, new channel value {}, outgoing {}, channel_id {}",
+	// 		self.channel_value_satoshis, is_outgoing, self.channel_id,
+	// 	);
+	// }
+
+	/// Get the splice message that can be sent during splice initiation.
+	#[cfg(splicing)]
+	pub fn get_splice_init(&self, our_funding_contribution_satoshis: i64,
+		funding_feerate_perkw: u32, locktime: u32,
+	) -> msgs::SpliceInit {
+		// Reuse the existing funding pubkey, in spite of the channel value changing
+		// (though at this point we don't know the new value yet, due tue the optional counterparty contribution)
+		// Note that channel_keys_id is supposed NOT to change
+		let funding_pubkey = self.get_holder_pubkeys().funding_pubkey.clone();
+		msgs::SpliceInit {
+			channel_id: self.channel_id,
+			funding_contribution_satoshis: our_funding_contribution_satoshis,
+			funding_feerate_perkw,
+			locktime,
+			funding_pubkey,
+			require_confirmed_inputs: None,
+		}
```

Comment:
Ah, the spec says `SHOULD` be different. https://github.com/lightning/bolts/pull/1160/files/aece0d8f42310d7e7d128905cf18eb0414c02459#r1631039188

@TheBlueMatt Are we fine with reusing the funding pubkey here?

Reply:
We can handle creating new funding pubkey separately.